Remove red check mark when editing metadata

lkiesow opened the pull request:
When editing meta data in the admin UI, a red check mark is shown on
changed (dirty/unwritten) fields. Red is not distinguishable from green
for some people, and the check mark as an indicator for unsaved changes
is not obvious.

Since the check marks do not mean anything anymore except that the user
finished their entry (which they should obviously know), this pull
request removes the check marks entirely.

If users try leaving the dialog with unsaved changes, a warning is
displayed anyway.

This fixes #3738

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

lkiesow committed:
Remove red check mark when editing metadata

When editing meta data in the admin UI, a red check mark is shown on
changed (dirty/unwritten) fields. Red is not distinguishable from green
for some people, and the check mark as an indicator for unsaved changes
is not obvious.

Since the check marks do not mean anything anymore except that the user
finished their entry (which they should obviously know), this pull
request removes the check marks entirely.

If users try leaving the dialog with unsaved changes, a warning is
displayed anyway.

This fixes #3738

mtneug commented:
The opinion in the dev meeting seems to be to have some way of identifying fields that changed before hitting save. Is the color important here as this only indicates changed fields?

KatrinIhler reviewed (dismissed):
I actually don't agree with removing this entirely, I think it's good to know which fields are gonna be changed once I hit save, if I edited multiple. You're of course right that red and green are indistinguishable for some people though (although I don't think the color matters terribly in this case). For more accessibility, I would suggest to also change the icon for the red marks instead.

lkiesow commented:
Personally, I would also say that red is usually a color indicating an error or danger. That's not the case here.

And really, you cannot remember what fields you edited likely in the last 60 seconds? Plus, that's not what the checkbox does right now. If that was the intention for this, it's already completely broken since it e.g. doesn't indicate if you remove a value.

KatrinIhler commented:
I'm also fine with changing the color, I'm not at all attached to red, and you're right that it probably isn't the greatest color for this. Maybe blue instead?

I think it's nice if you edit a field by mistake, or you get interrupted while editing, or you do a lot of changes all at once...

This was definitely the intention for the check mark, and if I remember correctly, this used to work. It should definitely also show when removing values. If it doesn't, it is indeed broken. :/

lkiesow committed:
Neutral Modification Indicator

This patch adds a neutral modification indicator to the metadata editor in the admin interface. This uses the existing
mechanism to detect changes – it doesn't improve the algorithm, nor does it make it worse – but switches to a neutral
indicator for identifying changes from the read (Why read? Is it an error) check mark (Why a check mark? Do we want to
indicate success?) it was before.

lkiesow commented:
I've changed the patch. It does no longer remove the modification indicator from the metadata editor of the admin interface, but uses a neutral one instead. This uses the existing mechanism to detect changes – it doesn't improve the algorithm, nor does it make it worse – but switches to a neutral indicator for identifying changes from the red (Why red? Is it an error) check mark (Why a check mark? Do we want to indicate success?) it was before.

``
![Screenshot from 2022-06-05 14-46-26](https://user-images.githubusercontent.com/1008395/172051373-9844f788-f614-46e9-b9c9-df3957a52bc0.png)

gregorydlogan commented:
This seems like a reasonable change to me.  Gets the point across without using colour, or checkmarks.

KatrinIhler reviewed (approved):
I'm fine with this change.

KatrinIhler commented:
> (Why red? Is it an error) check mark (Why a check mark? Do we want to indicate success?)

The answer to both of those questions is "Adding a save button for metadata was a very rushed last-minute change before a feature freeze iirc and I was so happy it worked that I didn't really consider the details" :P Keeping the check mark and simply changing the color (and red = unsaved) seemed like the easiest choice. So please, don't blame me anymore T_T
